### ISSUE TITLE
Registering Email fails with "undefined method `[]' for nil:NilClass"

### ISSUE THREAD

**Issue by LukasSchulz** [zammad/zammad#3048]
### Infos:

* Used Zammad version: 3.3.x
* Installation method (source, package, ..): package
* Operating system: Debian Buster
* Database + version:
* Elasticsearch version:
* Browser + version: Firefox

### Expected behavior:

*Registering a new Email in Settings works

### Actual behavior:

*The First step of the Email registering works: it replies, that X Emails are found on the server and will be moved, after that, the smtp check seems to work to, but the verification of sending an receiving fails with an error in the webgui ```undefined method `[]' for nil:NilClass```

### Steps to reproduce the behavior:

* Setting up an Email Account in the Settings.

More info to our setup:
Email setup works with another email provider, the thing that fails is our own email server using maddy (https://github.com/foxcpp/maddy)

And here an exerpt of the log:
```
I, [2020-05-09T10:20:41.120591 #28954-47068773302560]  INFO -- : Started GET "/api/v1/channels_email?_=1589017620179" for 10.1.2.201 at 2020-05-09 10:20:41 +0000
I, [2020-05-09T10:20:41.123707 #28954-47068773302560]  INFO -- : Processing by ChannelsEmailController#index as JSON
I, [2020-05-09T10:20:41.123738 #28954-47068773302560]  INFO -- :   Parameters: {"_"=>"1589017620179"}
I, [2020-05-09T10:20:41.141768 #28954-47068773302560]  INFO -- : Completed 200 OK in 18ms (Views: 2.5ms | ActiveRecord: 2.0ms)
I, [2020-05-09T10:20:41.143874 #28954-47068773302280]  INFO -- : Started POST "/api/v1/channels_email_inbound" for 10.1.2.201 at 2020-05-09 10:20:41 +0000
I, [2020-05-09T10:20:41.147065 #28954-47068773302280]  INFO -- : Processing by ChannelsEmailController#inbound as JSON
I, [2020-05-09T10:20:41.147102 #28954-47068773302280]  INFO -- :   Parameters: {"adapter"=>"imap", "options"=>{"host"=>"mail.infra4future.de", "user"=>"help@infra4future.de", "passwor
d"=>"[FILTERED]", "ssl"=>true, "port"=>"993", "folder"=>"", "keep_on_server"=>false}}
I, [2020-05-09T10:20:41.150849 #28954-47068773302280]  INFO -- : fetching imap (mail.infra4future.de/help@infra4future.de port=993,ssl=true,starttls=false,folder=INBOX,keep_on_server=
false)
I, [2020-05-09T10:20:41.581452 #28954-47068773302280]  INFO -- : check only mode, fetch no emails
I, [2020-05-09T10:20:41.779713 #28954-47068773302280]  INFO -- : Completed 200 OK in 633ms (Views: 0.1ms | ActiveRecord: 1.0ms)
I, [2020-05-09T10:20:42.961616 #28958-46990412998100]  INFO -- : Scheduler running...
I, [2020-05-09T10:20:42.966789 #28958-46990412998100]  INFO -- : Running job thread for 'Import OTRS diff load' (Import::OTRS.diff_worker) status is: sleep
I, [2020-05-09T10:20:42.966834 #28958-46990412998100]  INFO -- : Running job thread for 'Check Channels' (Channel.fetch) status is: sleep
I, [2020-05-09T10:20:42.982109 #28958-46990489379460]  INFO -- : execute Ticket.process_auto_unassign (try_count 0)...
I, [2020-05-09T10:20:42.983546 #28958-46990489379460]  INFO -- : ended Ticket.process_auto_unassign took: 0.016249905 seconds.
I, [2020-05-09T10:20:47.300541 #28954-47068717661020]  INFO -- : Started POST "/api/v1/channels_email_outbound" for 10.1.2.201 at 2020-05-09 10:20:47 +0000
I, [2020-05-09T10:20:47.303610 #28954-47068717661020]  INFO -- : Processing by ChannelsEmailController#outbound as JSON
I, [2020-05-09T10:20:47.303647 #28954-47068717661020]  INFO -- :   Parameters: {"adapter"=>"smtp", "options"=>{"host"=>"mail.infra4future.de", "user"=>"help@infra4future.de", "passwor
d"=>"[FILTERED]", "port"=>"465"}, "email"=>"help@infra4future.de"}
I, [2020-05-09T10:20:47.746813 #28954-47068717661020]  INFO -- : Completed 200 OK in 443ms (Views: 0.1ms | ActiveRecord: 0.9ms)
I, [2020-05-09T10:20:47.828770 #28954-47068773302560]  INFO -- : Started POST "/api/v1/channels_email_verify" for 10.1.2.201 at 2020-05-09 10:20:47 +0000
I, [2020-05-09T10:20:47.832016 #28954-47068773302560]  INFO -- : Processing by ChannelsEmailController#verify as JSON
I, [2020-05-09T10:20:47.832076 #28954-47068773302560]  INFO -- :   Parameters: {"inbound"=>{"adapter"=>"imap", "options"=>{"host"=>"mail.infra4future.de", "user"=>"help@infra4future.d
e", "password"=>"[FILTERED]", "ssl"=>true, "port"=>"993", "folder"=>"", "keep_on_server"=>false}}, "outbound"=>{"adapter"=>"smtp", "options"=>{"host"=>"mail.infra4future.de", "user"=>
"help@infra4future.de", "password"=>"[FILTERED]", "port"=>"465"}, "email"=>"help@infra4future.de"}, "meta"=>{"realname"=>"Infra4future", "email"=>"help@infra4future.de", "password"=>"
[FILTERED]", "group_id"=>"1"}, "group_id"=>"1"}
I, [2020-05-09T10:20:52.967335 #28958-46990412998100]  INFO -- : Running job thread for 'Process escalation tickets' (Ticket.process_escalation) status is: sleep
I, [2020-05-09T10:20:52.967479 #28958-46990412998100]  INFO -- : Running job thread for 'Generate Session data' (Sessions.jobs) status is: sleep
I, [2020-05-09T10:20:52.967512 #28958-46990412998100]  INFO -- : Running job thread for 'Check streams for Channel' (Channel.stream) status is: sleep
I, [2020-05-09T10:20:52.968159 #28958-46990412998100]  INFO -- : Running job thread for 'Execute jobs' (Job.run) status is: sleep
I, [2020-05-09T10:20:53.273466 #28954-47068773302560]  INFO -- : fetching imap (mail.infra4future.de/help@infra4future.de port=993,ssl=true,starttls=false,folder=INBOX,keep_on_server=
false)
```

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by MrGeneration** [comment]
Please provide the complete traceback of the error. All you provided is the informational part of the log - are there any errors?

**Comment by LukasSchulz** [comment]
That's the weird thing - I can't find anything in the production.log file - are there any other logs to look at?

The only error I see is this message in the Webgui:
<img width="567" alt="Screenshot 2020-05-09 at 14 56 27" src="https://user-images.githubusercontent.com/20911643/81474388-6417e200-9205-11ea-83b7-ead0ca727a12.png">

**Comment by LukasSchulz** [comment]
And that message only appears *after* I setup smtp too and zammad is trying to verify the setup - It doesn't fail to connect in the first place

Additional info after some testing: I tried the email address itself on apple mail and everything works fine, but when using fetchmail it outputs the following:
```
fetchmail: 6.4.0.beta4 querying mail.infra4future.de (protocol IMAP) at Sat 09 May 2020 12:51:14 PM UTC: poll started
Trying to connect to 46.4.15.78/143...connected.
fetchmail: IMAP< * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR STARTTLS LOGINDISABLED APPENDLIMIT COMPRESS] IMAP4rev1 Service Ready
fetchmail: IMAP> A0001 CAPABILITY
fetchmail: IMAP< * CAPABILITY IMAP4rev1 LITERAL+ SASL-IR STARTTLS LOGINDISABLED APPENDLIMIT COMPRESS
fetchmail: IMAP< A0001 OK CAPABILITY completed
fetchmail: IMAP> A0002 STARTTLS
fetchmail: IMAP< A0002 OK Begin TLS negotiation now
fetchmail: Loaded OpenSSL library 0x1010104f newer than headers 0x1010101f, trying to continue.
fetchmail: Server certificate:
fetchmail: Issuer Organization: Let's Encrypt
fetchmail: Issuer CommonName: Let's Encrypt Authority X3
fetchmail: Subject CommonName: mail.infra4future.de
fetchmail: Subject Alternative Name: mail.infra4future.de
fetchmail: mail.infra4future.de key fingerprint: 7B:64:1B:E5:C3:35:23:FE:1F:8F:9B:BE:1A:CC:0D:31
fetchmail: SSL/TLS: using protocol TLSv1.3, cipher TLS_AES_128_GCM_SHA256, 128/128 secret/processed bits
fetchmail: IMAP> A0003 CAPABILITY
fetchmail: IMAP< * CAPABILITY IMAP4rev1 LITERAL+ SASL-IR AUTH=PLAIN APPENDLIMIT COMPRESS
fetchmail: IMAP< A0003 OK CAPABILITY completed
fetchmail: mail.infra4future.de: upgrade to TLS succeeded.
fetchmail: IMAP> A0004 LOGIN "help@infra4future.de" *
fetchmail: IMAP< A0004 OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR APPENDLIMIT MOVE CHILDREN SPECIAL-USE COMPRESS UNSELECT IDLE] LOGIN completed
fetchmail: IMAP> A0005 SELECT "INBOX"
fetchmail: IMAP< * FLAGS (\Seen \Answered \Flagged \Deleted \Draft \Recent)
fetchmail: IMAP< * OK [PERMANENTFLAGS (\Seen \Answered \Flagged \Deleted \Draft \* \Recent)] Flags permitted.
fetchmail: IMAP< * OK [UNSEEN 1] Message 1 is first unseen
fetchmail: IMAP< * 29 EXISTS
fetchmail: IMAP< * 29 RECENT
fetchmail: IMAP< * OK [UIDNEXT 31] Predicted next UID
fetchmail: IMAP< * OK [UIDVALIDITY 2460576249] UIDs valid
fetchmail: IMAP< A0005 OK [READ-WRITE] SELECT completed
fetchmail: IMAP> A0006 EXPUNGE
fetchmail: IMAP< A0006 OK EXPUNGE completed
fetchmail: IMAP> A0007 SEARCH UNSEEN
fetchmail: IMAP< * SEARCH 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
fetchmail: IMAP< A0007 OK SEARCH completed
29 messages (1 seen) for help@infra4future.de at mail.infra4future.de.
fetchmail: IMAP> A0008 FETCH 1:29 RFC822.SIZE
fetchmail: IMAP< * 1 FETCH (RFC822.SIZE 1787)
fetchmail: IMAP< * 2 FETCH (RFC822.SIZE 1749)
fetchmail: IMAP< * 3 FETCH (RFC822.SIZE 9340)
fetchmail: IMAP< * 4 FETCH (RFC822.SIZE 5438)
fetchmail: IMAP< * 5 FETCH (RFC822.SIZE 5673)
fetchmail: IMAP< * 6 FETCH (RFC822.SIZE 3951)
fetchmail: IMAP< * 7 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 8 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 9 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 10 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 11 FETCH (RFC822.SIZE 1814)
fetchmail: IMAP< * 12 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 13 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 14 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 15 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 16 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 17 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 18 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 19 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 20 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 21 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 22 FETCH (RFC822.SIZE 1818)
fetchmail: IMAP< * 23 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 24 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 25 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 26 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 27 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< * 28 FETCH (RFC822.SIZE 1819)
fetchmail: IMAP< A0008 OK FETCH completed
fetchmail: IMAP> A0009 FETCH 1 RFC822.HEADER
fetchmail: IMAP< * 1 FETCH (BODY[HEADER] {1737}
fetchmail: IMAP< Delivered-To: help@infra4future.de
fetchmail: IMAP< Return-Path: <xxx@disroot.org>
fetchmail: IMAP< Authentication-Results: ; dkim=pass header.d=disroot.org
fetchmail: IMAP<  header.i=@disroot.org; spf=pass reason="matched 'a'"
fetchmail: IMAP<  smtp.helo=knopi.disroot.org smtp.mailfrom=disroot.org; dmarc=pass
fetchmail: IMAP<  header.from=disroot.org
fetchmail: IMAP< Received: from knopi.disroot.org (knopi.disroot.org [178.21.23.139]) by
fetchmail: IMAP<  mail.infra4future.de (envelope-sender <hexchen@disroot.org>) with ESMTPS id
fetchmail: IMAP<  dd181e08; Wed, 08 Apr 2020 17:34:01 +0000
fetchmail: IMAP< Received: from localhost (localhost [127.0.0.1])
fetchmail: IMAP<        by disroot.org (Postfix) with ESMTP id 3800D38D13
fetchmail: IMAP<        for <help@infra4future.de>; Wed,  8 Apr 2020 19:34:01 +0200 (CEST)
fetchmail: IMAP< X-Virus-Scanned: Debian amavisd-new at disroot.org
fetchmail: IMAP< Received: from knopi.disroot.org ([127.0.0.1])
fetchmail: IMAP<        by localhost (disroot.org [127.0.0.1]) (amavisd-new, port 10024)
fetchmail: IMAP<        with ESMTP id YX8q2-TJ3d0N for <help@infra4future.de>;
fetchmail: IMAP<        Wed,  8 Apr 2020 19:34:00 +0200 (CEST)
fetchmail: IMAP< From: xxx <xxx@disroot.org>
fetchmail: IMAP< DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=disroot.org; s=mail;
fetchmail: IMAP<        t=1586367240; bh=YOk2vtidvZxET4gEIP251d+UyvIeKdFLzlLV3QWzIwc=;
fetchmail: IMAP<        h=From:Subject:Date:To;
fetchmail: IMAP<        b=FpSJ9yuNYIShFJznXQ2OIR6J7CoQCgg8yDYEHIrPy/OLAaDuKRghSpwKPTE5bwmcB
fetchmail: IMAP<         ZRpX9Np1zfpA9JSAZTz8fKo+qTz/N/fjmdssn1KD8g9+Lnka2flE6jQj73tAR+WTzD
fetchmail: IMAP<         aHerRz3IchemIxqlEqIti1r9llktXaTMcxHLJeiC4XolVVvm04BYG7IQ1UX998OaUE
fetchmail: IMAP<         rDuz200RiuOfyackwhCsFN97ob9PnFmpl4k6Ov9cbVPdn3RXEhZUBftZkUuXdpw8/4
fetchmail: IMAP<         qqyue2N4soBv5e/Ea/v2kLeSEqkP3XybdJ552vbunh4Xi3mQ+vkrMlyiXkZVnyBLW1
fetchmail: IMAP<         Kbo59j2GH+clg==
fetchmail: IMAP< Content-Type: text/plain;
fetchmail: IMAP<        charset=us-ascii
fetchmail: IMAP< Content-Transfer-Encoding: 7bit
fetchmail: IMAP< Mime-Version: 1.0
fetchmail: IMAP< Subject: Help Me Please
fetchmail: IMAP< Message-Id: <FC3B18C6-4ADC-457F-9444-07A5F02802E8@disroot.org>
fetchmail: IMAP< Date: Wed, 8 Apr 2020 19:33:58 +0200
fetchmail: IMAP< To: help@infra4future.de
fetchmail: IMAP< 
fetchmail: IMAP< )
fetchmail: IMAP< A0009 OK FETCH completed
fetchmail: Incorrect FETCH response: * 1 FETCH (BODY[HEADER] {1737}.
fetchmail: IMAP> A0010 LOGOUT
fetchmail: IMAP< * BYE Closing connection
fetchmail: IMAP< A0010 OK LOGOUT completed
fetchmail: client/server synchronization error while fetching from help@infra4future.de@mail.infra4future.de
fetchmail: 6.4.0.beta4 querying mail.infra4future.de (protocol IMAP) at Sat 09 May 2020 12:51:15 PM UTC: poll completed
fetchmail: Query status=7 (ERROR)
fetchmail: normal termination, status 7
```

**Comment by MrGeneration** [comment]
Sorry but this looks like a setup issue on your mail server.
Ensure fetching, sending and deletion does work.

If it doesn't, it's for sure a configuration issue and no Zammad bug.
By default setting Zammad deletes mails it fetched from the inbox.

At this moment this sounds more like a technical question to me.

**Comment by thorsteneckel** [comment]
Would be really interesting where this issue comes from and provide a proper error message to the user. Any chance you can reproduce this on one of our [free hosted instances](https://zammad.com/getting-started) and give us the name?

**Comment by LukasSchulz** [comment]
I just registered an instance at infra4future.zammad.com and did all the steps I mentioned and received the same error.

**Comment by thorsteneckel** [comment]
Hey @LukasSchulz - thanks for trying it out. I extended a location where I expect the error to happen with an additional log line. Would you mind giving it another try? This will hopefully help us to narrow the root cause down.

**Comment by LukasSchulz** [comment]
Sure, did it again.

**Comment by thorsteneckel** [comment]
Thanks! I got the right location. Here is the traceback to the error message:

```
E, [2020-05-11T17:09:00.737605 #21111-38385580] ERROR -- : undefined method `[]' for nil:NilClass (NoMethodError)
/opt/zammad/app/models/channel/driver/imap.rb:179:in `block (2 levels) in fetch'
/opt/zammad/app/models/channel/driver/imap.rb:517:in `block in timeout'
/path/rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/timeout.rb:93:in `block in timeout'
/path/rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/timeout.rb:33:in `block in catch'
/path/rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/timeout.rb:33:in `catch'
/path/rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/timeout.rb:33:in `catch'
/path/rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/timeout.rb:108:in `timeout'
/opt/zammad/app/models/channel/driver/imap.rb:516:in `timeout'
/opt/zammad/app/models/channel/driver/imap.rb:178:in `block in fetch'
/opt/zammad/app/models/channel/driver/imap.rb:175:in `each'
/opt/zammad/app/models/channel/driver/imap.rb:175:in `fetch'
/opt/zammad/lib/email_helper/verify.rb:87:in `block in email'
/opt/zammad/lib/email_helper/verify.rb:76:in `times'
/opt/zammad/lib/email_helper/verify.rb:76:in `email'
/opt/zammad/app/controllers/channels_email_controller.rb:108:in `verify'
```

Zammad is at https://github.com/zammad/zammad/commit/008ca499c440fb2786ef7b4f0a9abf9c6309af77

So the error is thrown here:
https://github.com/zammad/zammad/blob/stable/app/models/channel/driver/imap.rb#L179

This tells me that there is an error when trying to fetch the (ENVELOPE) data of the message_id that of the verification mail. Seems like nothing gets returned here. That's strange. We'd need to further investigate this but it looks like an issue/incompatibility with (Ruby's Net::IMAP and) your mail server.

**Comment by LukasSchulz** [comment]
As i previosely stated, we are using maddy as our mail server (https://github.com/foxcpp/maddy). I am fairly certain that maddy implements everything as spec, but who knows 🤷

**Comment by eduardolagares** [comment]
I'm having the same problem here using umbler.com with IMAP.

**Comment by gdandersson** [comment]
Same here, for POP3. Zammad on Ubuntu Focal, running exim4 for SMTP. POP3 seems to be Dovecot.

`E, [2021-11-26T15:51:28.146207 #38231-174520] ERROR -- : undefined method `downcase' for nil:NilClass (NoMethodError)
lib/email_helper/probe.rb:291:in `outbound'
lib/email_helper/verify.rb:62:in `email'
app/controllers/channels_email_controller.rb:105:in `verify'
app/controllers/application_controller/has_download.rb:21:in `block (4 levels) in <module:HasDownload>'
app/controllers/application_controller/has_download.rb:20:in `block (3 levels) in <module:HasDownload>'
app/controllers/application_controller/has_download.rb:19:in `block (2 levels) in <module:HasDownload>'
app/controllers/application_controller/handles_transitions.rb:16:in `handle_transaction'
`

**Comment by martini** [comment]
@gdandersson Thanks for feedback. I use a simular setup (without issues). So may I ask for an example email account where I can verify this issue? support at zammad dot com with an reference to this issue would be fine!

**Comment by MrGeneration** [comment]
Thanks to @gdandersson we could have a look.
However, with the provided login credentials we could not reproduce this issue on a current 5.0.2 installation.

We've verified this on self hosted package and SaaS.
Please ensure to keep your installation up to date.

---

If you still believe we're wrong, please provide **full** information on your installation (meaning at least the following) as other wise we can not provide more time looking into this.

Thank you for your understanding.

```
- Used Zammad version: 
- Installation method (source, package, ..): 
- Operating system and its version:
- Database + version:
- Elasticsearch version:
- used E-Mail-Server (Name and version, per protocol if needed): 
- tried protocol (e.g. POP3 / IMAP):
```

**Comment by gdandersson** [comment]
- Used Zammad version: 5.0.2-1638167336.d2f24fc0.focal
- Installation method (source, package, ..): package (apt)
- Operating system and its version: Ubuntu 20
- Database + version: PostgreSQL 12.9
- Elasticsearch version: 7.15.2
- used E-Mail-Server (Name and version, per protocol if needed): Dovecot, unsure about version
- tried protocol (e.g. POP3 / IMAP): POP

Just after Zammad does it's "testing"
![Screenshot 2021-11-30 at 13 07 20](https://user-images.githubusercontent.com/52959348/144045949-0777a99d-3022-4bdb-be53-cc050fad48e5.jpg)

I then switched to POP3
![Screenshot 2021-11-30 at 13 07 31](https://user-images.githubusercontent.com/52959348/144045946-b2bdaa1d-7f35-4e24-b1e6-5981433d4ba9.jpg)

And this is the error, which actually looks .. good, since it says OK
![Screenshot 2021-11-30 at 13 07 36](https://user-images.githubusercontent.com/52959348/144045951-1a77c9d9-36d0-4d92-8c91-ffd85505791c.jpg)

**Comment by MrGeneration** [comment]
Port 993 is IMAP. I've verified this on Port 995 and it worked.
Please note that this is a technical question.

**Comment by gdandersson** [comment]
Ah, I assumed it would change port when I changed protocol.

Tried with 995 now and it found 2 emails, which would be moved.

A new popup showed:
StatusCode: 500
{"error":"undefined method `downcase' for nil:NilClass"}

Screenshot of JS-error:
<img width="468" alt="Screenshot 2021-11-30 at 13 24 22" src="https://user-images.githubusercontent.com/52959348/144047602-ab27dfc9-a444-4cd5-ac27-9a3807161f7a.png">

production.log shows exactly the same error as previously: undefined method `downcase' for nil:NilClass (NoMethodError)

This seems like a bug/issue to me, but maybe I'm missing something obvious?

**Comment by cautopartes** [comment]
I took this issue as an opportunity to set up a maddy instance to receive and deliver my Zammad test emails. 

Everything is working flawlessly, so I assume it is/was indeed not a Zammad bug.

**Comment by floatingpurr** [comment]
Same problem here when setting up an email account with IMAPS (imaps.aruba.it port 993 using TLS) 

```
zammad-docker-compose-zammad-railsserver-1    | E, [2023-01-04T13:10:42.133703 #1-112280] ERROR -- : undefined method `downcase' for nil:NilClass (NoMethodError)
```

Web Client gets 500 and gets stuck.

- Used Zammad version: 5.3.1-6
- Installation method (source, package, ..): Docker (https://github.com/zammad/zammad-docker-compose)
- Operating system and its version: Ubuntu 22.04 LTS
- Database + version: postgres:15.1-alpine
- Elasticsearch version: bitnami/elasticsearch:8.5.1
- used E-Mail-Server (Name and version, per protocol if needed): imaps.aruba.it port 993 using TLS
- tried protocol (e.g. POP3 / IMAP): IMAP (same for POP3)

**Comment by MrGeneration** [comment]
> Same problem here when setting up an email account with IMAPS 

different issue, look at the error message. Please don't hijack issues.

**Comment by floatingpurr** [comment]
I'm sorry, I didn't definitely want to hijack issues. I posted here, since I saw an identical error [some comments ago](https://github.com/zammad/zammad/issues/3048#issuecomment-980073483). I'll open up a new issue for my problem. Thanx and sorry again.

**Comment by MrGeneration** [comment]
Right. The original error is a different one. You're right someone else already hijacked so sorry that you were the first recipient of that line. We should keep issues clean because other wise we're risking a developer getting confused and possibly only fixing a specific portion of the issue.

If you can reliably reproduce the issue and ensure that we have an possibility to reproduce it as well then please create an independent issue. Thank you.

**Comment by floatingpurr** [comment]
No worries. I'm pretty sure there is a problem somewhere with the interaction with such an email server. I totally understand your point. I'll open up a new issue iif I'll be totally sure that I can share with you an email account of my company, according to the internal policy. Otherwise I'm afraid you wouldn't be able to look into it.

**Comment by cwooldridge** [comment]
If anyone else finds themselves with this error, for us we found that there was an email in the inbox that had a uid but there was not an email header and body to be fetched.  Returning an empty [] for header information.   Ideally zammad would skip this message.  We were able to discover which email by connecting to the IMAP inbox using the Thunderbird email client.  It told us the email id that it could not fetch.   We were then able to connect to the mail box using c# with MailKit and remove the id.  Once the email was removed everything worked as expected.
